### ISSUE TITLE
sudoer lines order is not preserved

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#23095]
When constructing a sudoers file we currently collect all sudoer lines in a map which leads to them being put in the file not in the order they appear in the role:

https://github.com/gravitational/teleport/blob/v12.1.0/lib/services/role.go#L2685

Order of sudo lines matters so we should preserve the order when writing a sudoers file. In case of multiple roles, we can write the sudoers files for each role in the user's role set (or in the username-role format) instead of a single file for the username?

**Comment by r0mant** [comment]
@jakule @lxea What do you folks think?

**Comment by jakule** [comment]
I don't think that the number of lines in sudoers file will be huge, so we could preserve the order by swapping the map to a slice in our implementation. I'm also unsure how creating a separate file per role would help with multiple roles. Someone needs to decide the correct order, which can be either us or sudo. From the docs https://linux.die.net/man/5/sudoers I see that sudoers are parsed in the lexicographical order:

> sudo will read each file in /etc/sudoers.d, skipping file names that end in '~' or contain a '.' character to avoid causing problems with package manager or editor temporary/backup files. Files are parsed in sorted lexical order. That is, /etc/sudoers.d/01_first will be parsed before /etc/sudoers.d/10_second. Be aware that because the sorting is lexical, not numeric, /etc/sudoers.d/1_whoops would be loaded after /etc/sudoers.d/10_second. Using a consistent number of leading zeroes in the file names can be used to avoid such problems.
 
I think we have two options here:
1. Do the same thing as sudo and sort roles by name (either in one or multiple files)
2. Add an optional field `priority` to each role and sort by that. 

The first option is simpler but forces people to create `10_create_user` role. The Second is probably cleaner
```yaml
name: create_user
...
  priority: 10
```
but introduces a new concept in the system.

**Comment by r0mant** [comment]
@jakule Right, my idea was to basically sort by role name to ensure order predictability without having to introduce another concept like priority. It can be the same file still, multiple files was just a way to have sudoers import logic do sorting itself.

**Comment by jakule** [comment]
We should also update our docs after this is implemented. Just adding it as an obvious note to not forget about it.
